sdcm/utils/k8s.py

```diff
@@ -609,6 +614,251 @@ def stop(self, timeout=None):
         self.join(timeout)
 
 
+class ScyllaPodsIPChangeTrackerThread(threading.Thread):
+    """It tracks changes of Scylla pods IP addresses in whole K8S cluster and updates mapping.
+
+    When a new IP is detected ...
+    - the mapping gets updated
+    - 'info' log message gets created
+    - 'info' event gets created.
+    - the monitoring node related to a Scylla pod gets reconfigured.
+
+    'mapper_dict' has following structure:
+    {
+        'scylla': {
+            'sct-cluster-dc-1-kind-0': {
+                'current_ip': '10.0.0.4',
+                'old_ips': ['10.0.0.2', '10.0.0.3'],
+            },
+            ...
+        },
+        'scylla-manager': {
+            'scylla-manager-manager-dc-manager-rack-0': {
+                'current_ip': '10.0.2.2',
+                'old_ips': [],
+            },
+            ...
+        },
+        ...
+    }
+    """
+
+    SCYLLA_PODS_SELECTOR = "app.kubernetes.io/name=scylla"
+    SCYLLA_PODS_EXPECTED_LABEL_KEYS = ("scylla/datacenter", "scylla/rack")
+
+    def __init__(self, k8s_kluster, mapper_dict):
+        self._termination_event = threading.Event()
+        super().__init__(daemon=True, name=self.__class__.name)
+        self.k8s_kluster = k8s_kluster
+        self.mapper_dict = mapper_dict
+        self.watcher = None
+        self._k8s_core_v1_api = KubernetesOps.core_v1_api(self.k8s_kluster.get_api_client())
+
+    @retrying(n=3600, sleep_time=1, allowed_exceptions=(ConnectionError, ))
+    def _open_stream(self, cache={}) -> None:  # pylint: disable=dangerous-default-value
+        try:
+            now = time.time()
+            if cache.get("last_call_at", 0) + 5 > now:
+                # NOTE: sleep for some time if it is called too often
+                time.sleep(5)
```

Comment:
wouldn't `sleep_time=5` in the retrying decorator would be enough ?

Reply:
It is not the reason for a `retry`. It must not be a retry from my POW.

Reply:
During the K8S platform upgrade, which longs for minutes, without this sleep we would send tons of requests to the unavilable API.

Reply:
Well this exact note is what missing here